lhotse/cut.py

```diff
@@ -741,6 +755,30 @@ def compute_offset():
             ))
         return CutSet.from_cuts(truncated_cuts)
 
+    def windows(self, duration: Seconds, keep_excessive_supervisions: bool = True) -> 'CutSet':
```

Comment:
Mm, OK, but this seems a rather non-ideal way to create cuts, because it doesn't make an attempt to avoid breaking supervisions up.  And surely we'd want to discard parts of the utterance that had no supervisions (or is that a separate operation)?

Reply:
It depends on the task, right? For VAD, every part of the recording can be useful as long as the supervisions tell us where the speech is. For ASR, we can have other ways of creating cuts - e.g. cut right at the supervision segment boundaries, or with a small leeway, or make equally sized cuts that don't break supervisions up. Each of those would be a method on the CutSet (to be added sometime soon).

It all assumes that the typical workflow would be: create RecordingSet and SupervisionSet (and maybe a FeatureSet for the whole recordings), then create a CutSet with Cuts that are the same duration as the Recordings, and then call a method (like `windows()`) on that CutSet to transform it in a specific way. The benefit of that approach is that the CutSet becomes the place where you can find all the necessary pieces of (meta)data, data dependencies, and the "standard" tools for building your dataset however you want.

Reply:
OK. Just bear it in mind when naming that function "windows".. if it's going to be just one of many, maybe choose a more specific name?